scipy/integrate/_tanhsinh.py

```diff
@@ -971,24 +987,30 @@ def _nsum(f, a, b, step=1, args=(), log=False, maxterms=int(2**20), atol=None,
     Returns
     -------
     res : _RichResult
-        An instance of `scipy._lib._util._RichResult` with the following
+        An object similar to an instance of `scipy.optimize.OptimizeResult` with the
         attributes. (The descriptions are written as though the values will be
         scalars; however, if `func` returns an array, the outputs will be
-
         arrays of the same shape.)
+
         success : bool
-            ``True`` when the algorithm terminated successfully (status ``0``).
-        status : int
+            ``True`` when the algorithm terminated successfully (status ``0``);
+            ``False`` otherwise.
+        status : int array
```

Comment:
What do you think about an Enum instead?

Reply:
Private (to replace variables like SIGNERR) or public? If the latter, where would it be imported from? Would it have its own documentation? What would users see when the result object is printed? Would users have to import it to index results by `status` (e.g. `res.nfev[res.status == -1]`

Reply:
In this case I suppose you would want to have an [IntEnum](https://docs.python.org/3/library/enum.html#enum.IntEnum). Users don't need to import it to make something like `res.status == -1`. If they print `res.status` though, it would tell them what it is.

```python
from enum import IntEnum
class Number(IntEnum):
    ONE = 1
    TWO = 2
    THREE = 3
    
Number.ONE == 1
# True
Number.ONE
# <Number.ONE: 1>
```

So self documented to me. For public/private, matter of choice I suppose. I would make it public. In libraries like Pydantic or REST frameworks things like these are public.

Reply:
> So self documented to me

Depends on the names. What would the names be to sum up these meanings, and what would they look like in an array?

Reply:
The fact that is an enum does not mean we could not document the values as you did here. Just saying there would also be a short tag as well for the enum itself, and this tag would help users in their code to see what it is without the need to open up the doc and go read that.

We can discuss this next week 😃

Reply:
Seen in person. Here is an attempt for names. But again, the is just a thought.


```python
from enum import IntEnum
class NsumStatus(IntEnum):
    converged = 0
    ab_invalid = -1
    iter_lim = -2
    non_finite = -3
    tol_lim = -4
```